pkg/controller/migration/convert/cni.go

```diff
@@ -1,14 +1,47 @@
 package convert
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
+	"io"
 	"log"
+	"net"
+	"reflect"
 	"strings"
 
 	"github.com/containernetworking/cni/libcni"
+	"github.com/containernetworking/cni/pkg/types"
 )
 
+// Used to parse the IPAM section out of the full CNI configuration
+type CNIHostLocalIPAM struct {
+	IPAM json.RawMessage `json:"ipam"`
+}
+
+// IPAMConfig represents the IP related network configuration.
+// This nests Range because we initially only supported a single
+// range directly, and wish to preserve backwards compatability
+type HostLocalIPAMConfig struct {
```

Comment:
Did you consider just importing this as well? https://github.com/containernetworking/plugins/blob/master/plugins/ipam/host-local/backend/allocator/config.go#L43-L52

Reply:
Yep the problem is the Subnet we use (`usePodCidr`) is not actually valid and the Calico CNI replaces that field when actually running. So I needed to be able to replace its definition.

Reply:
We already handle a replacement of `__CNI_MTU__` in the CNI parsing code. Is there a reason we don't want to reuse that approach for this one?